homeassistant/components/esphome/__init__.py

```diff
@@ -114,14 +114,16 @@ def async_on_service_call(service: HomeassistantServiceCall) -> None:
                     template.render_complex(data_template, service.variables)
                 )
             except TemplateError as ex:
-                _LOGGER.error("Error rendering data template: %s", ex)
+                _LOGGER.error("Error rendering data template for %s: %s", host, ex)
```

Comment:
Where is `host` set? I don't see it in this function scope...

Reply:
It is up in the begging of `async def async_setup_entry`, because it's a function-inside-a-function
L61-L67:
```python
async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
    """Set up the esphome component."""
    hass.data.setdefault(DATA_KEY, {})

    host = entry.data[CONF_HOST]  # Host from here <---
    port = entry.data[CONF_PORT]
    password = entry.data[CONF_PASSWORD]
    [...]
    
    @callback
    def async_on_service_call(service: HomeassistantServiceCall) -> None:
        [...]
        if service.is_event:
            if domain != "esphome":
                _LOGGER.error(
                    "Can only generate events under esphome domain! (%s)", host  # To here <---
                )
                return
```

Reply:
Sorry, I was on my morning coffee when I looked at this. 👍